lib/opam_solve.ml

```diff
@@ -52,6 +53,7 @@ module Opam_monorepo_context (Base_context : BASE_CONTEXT) :
     require_dune : bool;
     opam_provided : OpamPackage.Name.Set.t;
     require_cross_compile : bool;
+    preferred_versions : OpamTypes.version OpamPackage.Name.Map.t option;
```

Comment:
As mentioned in the other comments, maybe an empty map would be a cyclomatically less complex version.

I feel that there is no semantic difference in this case between no preferred versions in a map and no map - to me it feels like it expresses the same state ("no version preferred") but with the option there are now two ways to express this.